framework/src/media/audio/audio_manager.c

```diff
@@ -565,87 +672,139 @@ int start_audio_stream_out(void *data, uint32_t frames)
 audio_manager_result_t set_audio_stream_in(uint8_t channels, uint32_t sample_rate, uint8_t format)
 {
 	int card_id = -1;
+	uint32_t closest_samprate;
 	struct pcm_config config;
 
 	if ((channels <= 0) || (sample_rate <= 0) || (format < 0)) {
 		return AUDIO_MANAGER_INVALID_PARAM;
 	}
 
 	if ((g_audio_in_card_num == 0) && (find_input_audio_card() != AUDIO_MANAGER_SUCCESS)) {
-		auddbg("Found no active input audio card\n");
+		printf("Found no active input audio card\n");
 		return AUDIO_MANAGER_CARD_NOT_FOUND;
 	}
 
-	audvdbg("Print the current card id = %d\n", card_id);
+	printf("Print the current card id = %d\n", card_id);
 	card_id = get_avail_audio_in_card_id();
 	if (card_id < 0) {
-		auddbg("Found no available card\n");
+		printf("Found no available card\n");
 		return AUDIO_MANAGER_NO_AVAIL_CARD;
 	}
 
 	if (channels > AUDIO_STREAM_CHANNEL_MONO) {
-		config.channels = AUDIO_STREAM_CHANNEL_STEREO;
+		channels = AUDIO_STREAM_CHANNEL_STEREO;
 	}
+
+	closest_samprate = getClosestSampRateUp(sample_rate);
```

Comment:
we can use config.rate directly.

Reply:
config.rate is located below line.

Reply:
I mean you can modify to config.rate = getClosest...();

Reply:
You're right. Fixed